tensorflow_addons/text/crf.py

```diff
@@ -209,7 +213,7 @@ def crf_log_likelihood(
         transition_params = tf.Variable(
             initializer([num_tags, num_tags]), "transitions"
         )
-
+    transition_params = tf.cast(transition_params, inputs.dtype)
```

Comment:
I think that by a computational point of view probably it is quite the same so do you think that it makes sense to put in the `else` branch?

Reply:
I think current way is much safer when it talks about mixed precision. I figure out that this casting will maks the returned `transition_params` have different type from the input one.

BTW, from the aspect of design pattern, I do not think this function should accept `None` of `transition_params` as input, creates it and returns it. Users should prepare the `transition_params` in advance and call the function. This is how our `layers/crf_test.py` does.

Reply:
Yes probably.
I only meant that Variabile has dtype if we will maintain the if.

Reply:
@zara44 Just an opinion from @facaiy or @Squadrick about removing `None` case.

Reply:
@Squadrick any chance you can have a look at this? :)

Reply:
@facaiy are you able to look into this merge request? Thank you!

Reply:
@seanpmorgan are you able to review this?

Reply:
@bhack can you approve this first? I believe they are now busy at their work, so may not response recently. Seems that this PR might be helpful for the community, I would think we can merge this first as a quick fix for mixed precision. Any suggestion change would also be really appreciated. Thanks!

Reply:
Ok can you add an inline Todo to re-evaluate It later?

Reply:
Done 😃